driver/bpf/fillers.h

```diff
@@ -4873,10 +5034,30 @@ FILLER(sys_procexit_e, false)
 	}
 	CHECK_RES(res);
 
-	/* Did it produce a core? */
+	/* Parameter 4: core (type: PT_UINT8) */
 	res = bpf_push_u8_to_ring(data, __WCOREDUMP(exit_code) != 0);
 	CHECK_RES(res);
 
+	/* Parameter 5: reaper (type: PT_PID) */
+	/* This is a sort of optimization if this thread has no children in the kernel
+	 * we don't need a reaper and we can save some precious cycles.
+	 * We send `reaper_pid==0` if the userspace still has some children
+	 * it will manage them with its userspace logic.
+	 */	
+	pid_t reaper_pid = 0;
+	struct list_head *head = &(task->children);
+	struct list_head *next_child = (struct list_head *)_READ(head->next);
+	if(next_child != head)
+	{
+		/* We have at least one child, so we need a reaper for it */
+		reaper_pid = find_new_reaper_pid(data, task);
+	}
+	/* Please note here `pid` is in kernel-lingo so it is a thread id.
```

Comment:
For execve we have

```
/* Parameter 4: tid (type: PT_PID) */
pid_t pid = _READ(task->pid);
res = bpf_push_s64_to_ring(data, pid);
CHECK_RES(res);

/* Parameter 5: pid (type: PT_PID) */
pid_t tgid = _READ(task->tgid);
res = bpf_push_s64_to_ring(data, tgid);
CHECK_RES(res);
```

we all know how confusing and annoying this `tid` vs `pid` dilemma is in the kernel, since it's `tid` calling the variable `reaper_pid` is consistent, but then in the PPME definition we should call it `reaper_tid` instead of just `reaper` so we know what it is?

Reply:
done in the last commit, thanks :)